### PR TITLE
WIP: Generating invalid data

### LINKED ISSUE #65: Negative testing
The current approach produces data that **matches** the schema. To verify that wrong data will not be processed without and error we need to mutate the schema in runtime before creating a strategy in the way that it will not match the initial schema (e.g. it shouldn't be a subset of the initial schema). Could be another "test_type" option.

With this feature, we will expand the scope of errors that could be discovered by Schemathesis

### PR DESCRIPTION
**Pull request by Stranger6667**
Resolves #65 and #66 

- [ ] properly filter data, generated by not_from_schema - should be dictionaries and for path probably not arrays or objects
- [ ] high level tests for not valid data
- [x] CLI option for invalid data generation
- [ ] Support in CLI / Runner
- [ ] Support in lazy schemas
- [ ] changelog entry
- [ ] docs

### CODE REVIEW

File: src/schemathesis/_hypothesis.py [hunk 1]
```diff
@@ -107,38 +143,77 @@ def is_surrogate(item: Any) -> bool:
...
-                        from_schema(value).filter(filter_path_parameters).map(quote_all)  # type: ignore
-                    )
+                    strategies[parameter] = strategy.filter(filter_path_parameters).map(quote_all)  # type: ignore
                 elif parameter in ("headers", "cookies"):
-                    strategies[parameter] = from_schema(value).filter(is_valid_header)  # type: ignore
+                    strategies[parameter] = strategy.filter(is_valid_header)  # type: ignore
                 elif parameter == "query":
-                    strategies[parameter] = from_schema(value).filter(is_valid_query)  # type: ignore
+                    strategies[parameter] = strategy.filter(is_valid_query)  # type: ignore
+                elif parameter == "cookies":
```
Comment (barrett-schonefeld) [inline src/schemathesis/_hypothesis.py:211]: Do we want this block to execute? The `elif parameter in ("headers", "cookies")` will evaluate to True before we reach `elif parameter == "cookies"`.

Comment (Stranger6667) [reply src/schemathesis/_hypothesis.py:211]: Indeed! Will clean it up :) Thanks

At the moment I am not sure when I can get back to this pull request :(

Comment (barrett-schonefeld) [reply src/schemathesis/_hypothesis.py:211]: That's OK! I'm just sifting through this to review as part of reviewing how requests are generated. I'm going to start opening some issues for features (one of them will be modify requests before they are sent), and I will start taking issues off the board in about a week.

File: README.rst [hunk 2]
```diff
@@ -215,11 +215,12 @@ Each test function should have the ``case`` fixture, that represents a single te
 
 Important ``Case`` attributes:
 
-- ``method`` - HTTP method
-- ``formatted_path`` - full endpoint path
-- ``headers`` - HTTP headers
-- ``query`` - query parameters
-- ``body`` - request body
+- ``input_type`` - type of data in the case, it can be valid or invalid;
```
Comment (barrett-schonefeld) [inline README.rst:218]: What will the default be? Would like to have a way to generate both valid and invalid data in the same run of tests.

**Comment by Stranger6667** [comment]
Closed in favor of #831

**Pull request closed without merging**